### ISSUE TITLE
linter: Docs are incorrect for no-restricted-jest-methods

### ISSUE THREAD

**Issue by connorshea** [oxc-project/oxc#16733]
### What version of Oxlint are you using?

1.31.0

### What command did you run?

n/a

### What does your `.oxlintrc.json` config file look like?

n/a

### What happened?

The config option allows arbitrary option names, but the config docs suggest that this is supposed to be called `restrictedJestMethods`, which is wrong. https://github.com/oxc-project/oxc/blob/397bcd5f0febc04b8bed16f0a5a8dc80ca2e5469/crates/oxc_linter/src/rules/jest/no_restricted_jest_methods.rs

See the original rule and its docs: https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-restricted-jest-methods.md#options

**Comment by tt-a1i** [comment]
Confirmed - there's a mismatch between the JsonSchema-generated docs and the actual `from_configuration` implementation.

**What the docs say** (via JsonSchema):
```json
["error", { "restrictedJestMethods": { "fn": null } }]
```

**What actually works** (matching ESLint):
```json
["error", { "fn": null, "mock": "Do not use mocks" }]
```

The `from_configuration` implementation directly parses the first array element as an object:

```rust
fn from_configuration(value: serde_json::Value) -> Self {
    let restricted_jest_methods = &value
        .get(0)  // gets first element directly
        .and_then(serde_json::Value::as_object)  // treats it as the method map
        .map(Self::compile_restricted_jest_methods)
        // ...
}
```

The test cases confirm the ESLint-compatible format:
```rust
("jest.fn()", Some(serde_json::json!([{ "fn": null }]))),
("jest.mock()", Some(serde_json::json!([{ "mock": "Do not use mocks" }]))),
```

The fix would be to either:
1. Update the JsonSchema to match the actual config format, or
2. Add a custom schema that reflects the ESLint-style config

**Comment by Boshen** [comment]
Fixed on `main`. The `no-restricted-jest-methods` / `no-restricted-vi-methods` config schema was realigned with the implementation (#22920, #22921, #23246) — the generated schema no longer advertises a `restrictedJestMethods` wrapper key. It is now a flat map of method-name → (custom message | `null`), which is exactly what the rule accepts.

Verified on current `main`: the config `{"jest/no-restricted-jest-methods": ["error", {"advanceTimersByTime": "no fake timers please"}]}` parses correctly and the rule fires with the custom message. Closing as resolved.
